### DISCUSSION TITLE
Feature request: Local repo for remote database

### DISCUSSION THREAD

**Discussion by zdateh** [intersystems/git-source-control#271] [Ideas]
Hi! Loving the extension - it's enabled us to finally implement reliable source control!

Our use-case is Studio installed on multiple developer's local machines, connecting to single IRIS instance on a remote machine. We have unique namespaces on the remote machine for each developer.

This works ok with git-source-control - you can set up a repository folder on the remote machine folder each namespace (developer), and still work on your local machine, by issuing git commands via the new Studio Git functions.

It is however a bit irritating having to remote desktop to the remote machine to do Git housekeeping, or anything not handled by the current Studio Git functions. And also, as someone familiar with Git command line I prefer to work with it directly, rather than via a UI (nothing personal! 😄), so again it's a little painful to remote in.

Since Studio can export to your local file system (via Tools > Export... > Local), would it be possible for the extension to sync to/from a repo on the local file system, rather than only to the machine that IRIS is being hosted?

Thanks,
Chris

**Comment by isc-tleavitt** [comment]
@zdateh sorry this went so long without comment!

This isn't really in scope for what an "embedded source control" extension (this is my term) can do. An embedded source control extension presumes that the repo is on the server, and is largely concerned with taking changes made on the server (through Studio, isfs mode VSCode, and Management Portal editors) and properly reflecting them on the server's filesystem.

If you want to work with a developer's local copy of the repo (on their own machine) and handle getting that into a remote environment, *assuming that remote environment is dedicated to that developer's use and you're not using management portal editors for interoperability or analytics* (where you'd also want changes reflected in the repo), you might consider using VSCode with a client-centric mode. But if you want changes made to a developer's own filesystem to be treated the same as changes made via management portal editors, there isn't a seamless solution for that.

**Reply by zdateh** [reply]
Fair enough! 

I think it's a little bit of a shame, because I imagine most collaborative development use cases for IRIS are using Studio on local machine, and connecting to a remote IRIS database. That's possible now if you work only via the Studio git extension menu, but there's enough bugs and UI issues (eg #269) at the moment (I've not logged any more because I want to try the next version first!) that we just do all git operations via Git bash.

In my head, given that Studio itself is happy exporting database code to server file system OR local file system, the Git extension might work similarly, but it's probably not that simple :)

BUT once the extension matures, we presumably will be able to do everything via it.

VSCode is next on my list! :)
